### PR TITLE
Convert repo into a template repo

### PR DESCRIPTION
**Pull request by yuvipanda**
This PR merges in https://github.com/yuvipanda/rstudio-binder-template (preserving history),
mostly replacing what was here. The README content is different, as it encourages users to
use GitHub's new template repository feature (which wasn't around when rocker/binder was first
created) to create their own repos that will work on mybinder.org.

Based on the comment in https://github.com/rocker-org/binder/pull/46#issuecomment-998193333
by @cboettig.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,29 @@
+BSD 3-Clause License
```
Comment (eddelbuettel) [inline LICENSE]: Well we had this both on our website (from where the screenshot it taken) and most documents.  

image

Comment (yuvipanda) [reply LICENSE]: @eddelbuettel ah good catch! I've changed the license to GPL.

Comment (eddelbuettel) [reply LICENSE]: :heart:

Thanks for the quick update.  We're not _that_ dogmatic about but it is nice to be consistent about it esp given the top-level statement.

Comment (yuvipanda) [reply LICENSE]: Yeah absolutely. It wasn't intentional, I had just forgotten.

**Comment by cboettig** [comment]
:tada:

**Comment by eitsupi** [comment]
Maybe some settings are needed to make this repo to template?
https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository

**Comment by yuvipanda** [comment]
Yeah, someone with admin needs to check that flag. @cboettig maybe?

**Comment by cboettig** [comment]
yup, done!

**Comment by yuvipanda** [comment]
@cboettig \o/ ty

**Pull request merged**